spark/src/test/java/org/apache/iceberg/actions/TestRemoveOrphanFilesAction.java

```diff
@@ -569,4 +672,349 @@ public void testRemoveOrphanFilesWithHadoopCatalog() throws InterruptedException
         .collectAsList();
     Assert.assertEquals("Rows must match", records, actualRecords);
   }
+
+  @Test
+  public void testFindOrphanFilesWithSameAuthority() throws Exception {
+    List<Row> validFilesData = getRowsWithFilePath(HDFS_SERVICENAME_USER_LOG_DATA_DUMMY_FILE, 3);
+
+    Dataset<Row> validFileDS = RemoveOrphanFilesAction
+        .addRelativePathColumn(
+            spark.createDataset(validFilesData, RowEncoder.apply(constructStructureWithString())));
+
+    List<Row> actualFilesData = new ArrayList<>();
+
+    actualFilesData.addAll(validFilesData);
+    List<Row> expectedOrphanFiles = getRowsWithFilePath(HDFS_SERVICENAME_USER_LOG_DATA_ORPHAN_FILE, 4);
+    actualFilesData.addAll(expectedOrphanFiles);
+
+    Dataset<Row> actualFileDS = RemoveOrphanFilesAction
+        .addRelativePathColumn(
+            spark.createDataset(actualFilesData, RowEncoder.apply(constructStructureWithString())));
+
+    List<String> orphanFiles = RemoveOrphanFilesAction.findOrphanFiles(validFileDS, actualFileDS);
+
+    Assert.assertNotNull(orphanFiles);
+
+    Assert.assertEquals(expectedOrphanFiles.stream().map(row -> row.get(0).toString())
+        .collect(Collectors.toList()), orphanFiles);
+  }
+
+  @Test
+  public void testFindOrphanFilesWithValidFileHasNoAuthority() throws Exception {
+    List<Row> validFilesData = getRowsWithFilePath(HDFS_USER_LOG_DATA_DUMMY_FILE, 4);
+    Dataset<Row> validFileDS = RemoveOrphanFilesAction
+        .addRelativePathColumn(
+            spark.createDataset(validFilesData, RowEncoder.apply(constructStructureWithString())));
+
+    List<Row> actualFilesData = new ArrayList<>();
+
+    actualFilesData.addAll(getRowsWithFilePath(HDFS_SERVICENAME_USER_LOG_DATA_DUMMY_FILE, 4));
+    List<Row> expectedOrphanFiles = getRowsWithFilePath(HDFS_SERVICENAME_USER_LOG_DATA_ORPHAN_FILE, 4);
+    actualFilesData.addAll(expectedOrphanFiles);
+
+    Dataset<Row> actualFileDS = RemoveOrphanFilesAction
+        .addRelativePathColumn(
+            spark.createDataset(actualFilesData, RowEncoder.apply(constructStructureWithString())));
+
+    List<String> orphanFiles = RemoveOrphanFilesAction.findOrphanFiles(validFileDS, actualFileDS);
+
+    Assert.assertEquals(expectedOrphanFiles.stream().map(row -> row.get(0).toString())
+        .collect(Collectors.toList()), orphanFiles);
+  }
+
+  @Test
```

Comment:
My gut tells me we should be able to parameterize all of tests around the FileName so we can remove a bit of duplicate code here.

Reply:
done